### PR TITLE
update stats tests

### PR DESCRIPTION
**Pull request by dpo**
See
https://github.com/JuliaSmoothOptimizers/SolverCore.jl/pull/27
https://github.com/JuliaSmoothOptimizers/SolverCore.jl/pull/28

### CODE REVIEW

**Comment by dpo** [comment]
@abelsiqueira @tmigot The tests pass locally with the `main` version of SolverCore. This PR also sets the scene for pre-allocating `stats` outside of `solve!()`.

File: Project.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ SolverTools = "b5612192-2639-5dc1-abfe-fbedd65fab29"
 
 [compat]
 Krylov = "0.8"
-LinearOperators = "1.1, 2.0"
+LinearOperators = "2.0"
```
Comment (tmigot) [inline Project.toml:18]: Is there a reason why we drop 1.1 here?

File: src/lbfgs.jl [hunk 2]
```diff
@@ -188,13 +188,11 @@ function solve!(
...
-  return GenericExecutionStats(
-    status,
-    nlp,
-    solution = x,
-    objective = f,
-    dual_feas = ∇fNorm,
-    iter = iter,
-    elapsed_time = elapsed_time,
-  )
+  stats = GenericExecutionStats(status, nlp)
```
Comment (tmigot) [inline src/lbfgs.jl:191]: Should the constructor `GenericExecutionStats(status, nlp, kwargs...)` be deprecated now? Unless, we plan to modify it to set the `reliable` booleans.

**Review by tmigot — COMMENTED** [review]
It looks good to me, I just have minor comments.

**Review by abelsiqueira — APPROVED** [review]

File: src/lbfgs.jl (hunk 2 above)
Comment (dpo) [reply src/lbfgs.jl:191]: I introduced a simplified constructor in https://github.com/JuliaSmoothOptimizers/SolverCore.jl/pull/31.

However, the current constructor remains necessary for those who want special storage for the bounds multipliers (e.g., sparse vectors).

File: Project.toml (hunk 1 above)
Comment (dpo) [reply Project.toml]: Probably not. I restored it, thanks.

File: src/lbfgs.jl (hunk 2 above)
Comment (dpo) [reply src/lbfgs.jl:191]: @tmigot On @geoffroyleconte's recommendation, I removed the "simplified" constructor as it was simply redundant.

Comment (tmigot) [reply src/lbfgs.jl:191]: Awesome, thanks!

**Review by tmigot — APPROVED** [review]
Looks good to me, ready to merge after SolverCore.jl's new release.

**Pull request merged**